cli_client/python/timesketch_cli_client/cli.py

```diff
@@ -0,0 +1,146 @@
+# Copyright 2021 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Timesketch CLI client."""
+
+import sys
+import click
+
+from requests.exceptions import ConnectionError as RequestConnectionError
+
+from timesketch_api_client import config as timesketch_config
+
+from timesketch_cli_client.commands import analyze
+from timesketch_cli_client.commands import config
+from timesketch_cli_client.commands import importer
+from timesketch_cli_client.commands import search
+from timesketch_cli_client.commands import sketch as sketch_command
+from timesketch_cli_client.commands import timelines
+
+from .definitions import DEFAULT_OUTPUT_FORMAT
+from .version import get_version
+
+
+class TimesketchCli(object):
+    """Timesketch CLI state object.
+
+    Attributes:
+        sketch_from_flag: Sketch ID if provided by flag
+        config_assistant: Instance of ConfigAssistant
+    """
+    def __init__(self, api_client=None, sketch_from_flag=None, conf_file=None):
+        """Initialize the state object.
+
+        Args:
+            sketch_from_flag: Sketch ID if provided by flag.
+        """
+        self.api = api_client
+        self.sketch_from_flag = sketch_from_flag
+
+        if not api_client:
+            try:
+                self.api = timesketch_config.get_client(load_cli_config=True)
```

Comment:
ok.. so there are other options here, which I'm wondering whether you should support... mainly `config_section`, which can be used and is sometimes used if you have more than a single TS server to connect to (think production, dev, something).... and also `token_password` if you manually set the pwd for the token file.

Should these not be supported?  I Think so... but I can take a TODO for now

Reply:
TODO added